genie-web/src/main/java/com/netflix/genie/web/services/impl/LocalJobKillServiceImpl.java

```diff
@@ -108,9 +120,21 @@ public void killJob(
         @NotBlank(message = "No id entered. Unable to kill job.") final String id,
         @NotBlank(message = "No reason provided.") final String reason
     ) throws GenieException {
+
         // Will throw exception if not found
         // TODO: Could instead check JobMonitorCoordinator eventually for in memory check
         final JobStatus jobStatus = this.jobSearchService.getJobStatus(id);
+
+        //TODO possible optimization. Have a single projection for getting hostname and isV4Job.
+        //Pass isV4Job as a method parameter. That will save a db call.
+        final boolean isV4Job = jobPersistenceService.isV4(id);
```

Comment:
Do you use this anywhere else? any reason not to just have it in the `if()` directly?

Reply:
had restructrued the code and this variable was left out. Fixed